conans/model/recipe_ref.py

```diff
@@ -0,0 +1,29 @@
+from conan.api.output import ConanOutput
+from conan.internal.model.recipe_ref import RecipeReference as _RecipeReference
+
+_msg = """
+*******************************************************************
+private '{}'
+detected in user code (custom commands, extensions, recipes, etc).
+Stop using it, use only public documented APIs.
```

Comment:
Do we want to warn/scare users saying that we *will* break this usage in the future?

Reply:
Do you mean it is not scary enough? 😅 

I can explicit that it will break, sure.

Reply:
Done